### ISSUE TITLE
[Feature Request] For "Legacy Camera Lock" tweak, add option to disable tweak during auto-run

### ISSUE THREAD

**Issue by ryankhart** [Caraxi/SimpleTweaksPlugin#287]
I love the "Legacy Camera Lock" tweak. It's the primary reason why I install Simple Tweaks.

But one thing I miss about the unmodified game is when I turn on auto-run for running long distances before I unlock flying in a zone. In these circumstances, I'm often following winding road at a slow speed, and I tend to want to play one handed on controller/gamepad. But instead, I need to keep both my thumbs on the thumbsticks at all times.

It would be nice if there was an option, disabled by default, which disables the "Legacy Camera Lock" functionality only for auto-run, and then turn it back on when auto-run has ended or has been interrupted.

**Comment by Caraxi** [comment]
I can do that

**Comment by ryankhart** [comment]
Wow! This looks fantastic! I looked at the new code, and I learned a lot from it.

**Comment by ryankhart** [comment]
Heads up. I just built Simple Tweaks from source to test this out and it was great at first, but I encountered a bug where it didn't reactivate the "Legacy Camera Lock" tweak after auto-run stopped.

I'm still working on repro steps for the bug, but I thought I'd post this here at least before this change gets released in a stable release.

**Comment by ryankhart** [comment]
It appears to me that it works only for the first time that auto-run is enabled, but it does not un-disable "Legacy Camera Look" when ending auto-run.

A second issue is that when it does break, the "Legacy Camera Lock" tweak doesn't work at all, even when I unchecked and checked the box in the config to disable and re-enable it. I had to disable the plugin, reinstall the plugin, and uncheck and check the box next to "Disable While Auto Running" to get it working again.
